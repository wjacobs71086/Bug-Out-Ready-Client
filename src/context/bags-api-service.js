import config from '../config'
import TokenService from './token-service';

const BagsApiService = {
  getThings() {
    return fetch(`${config.API_ENDPOINT}/bags`, {
      headers: {
        'authorization': `bearer ${TokenService.getAuthToken()}`,
      },
    })
      .then(res =>
        (!res.ok)
          ? res.json().then(e => Promise.reject(e))
          : res.json()
      )
  },
  getThing(bagId) {
    console.log()
    return fetch(`${config.API_ENDPOINT}/bag-home/${bagId}`, {
      headers: {
        'authorization': `bearer ${TokenService.getAuthToken()}`,
      },
    })
      .then(res => {
        (!res.ok)
          ? res.json().then(e => Promise.reject(e))
          : res.json()
      })
  },
  getBagsItems(bagId) {
    console.log(`${config.API_ENDPOINT}/bag-home/${bagId}`);
    return fetch(`${config.API_ENDPOINT}/bag-home/${bagId}`, {
      headers: {
        'authorization': `bearer ${TokenService.getAuthToken()}`,
      },
    })
      .then(res =>
        (!res.ok)
          ? res.json().then(e => Promise.reject(e))
          : res.json()
      )
  }

}

export default BagsApiService;