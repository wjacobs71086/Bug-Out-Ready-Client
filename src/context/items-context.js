import React, { Component } from 'react';


const ItemsListContext = React.createContext({
  itemsList: [],
  bagsList: [],
  userId: [],
  error: null,
  setError: () => {},
  clearError: () => {},
  setItemsList: () => {},
  setBagsList: () => {},
})
export default ItemsListContext

export class ItemsListProvider extends Component {
  state = {
    itemsList: [],
    bagsList: [],
    error: null,
  };

  setItemsList = itemsList => {
    this.setState({ itemsList })
  }

  setBagsList = bagsList => {
    this.setState({ bagsList })
  }

  setError = error => {
    console.error(error)
    this.setState({ error })
  }

  clearError = () => {
    this.setState({ error: null })
  }

  render() {
    const value = {
      itemsList: this.state.itemsList,
      error: this.state.error,
      setError: this.setError,
      clearError: this.clearError,
      setItemsList: this.setItemsList,
      setBagsList: this.setBagsList,
      bagsList: this.state.bagsList,
      userId: this.state.userId,
    }
    return (
      <ItemsListContext.Provider value={value}>
        {this.props.children}
      </ItemsListContext.Provider>
    )
  }
}