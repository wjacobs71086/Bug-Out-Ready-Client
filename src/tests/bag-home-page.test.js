import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import BagHomePage from '../routes/bag-home-page/bag-home-page';
import { BrowserRouter } from 'react-router-dom'
import { ItemsListProvider } from '../context/items-context';
import { Route
} from 'react-router-dom'

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <BrowserRouter>
            <ItemsListProvider>
                <Route
                    to
                    component={BagHomePage}
                />
            </ItemsListProvider>
        </BrowserRouter>
        , div);
    ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
    const tree = renderer
      .create(        
      <BrowserRouter>
        <ItemsListProvider>
            <Route
                to
                component={BagHomePage}
            />
        </ItemsListProvider>
    </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
});
