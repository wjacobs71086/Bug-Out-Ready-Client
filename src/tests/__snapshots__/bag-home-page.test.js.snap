// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div>
  <div
    className="headerLinks"
  >
    <div
      className="logoutLink"
    >
      <a
        href="/login"
        id="logoutLink"
        onClick={[Function]}
      >
        Logout
      </a>
    </div>
    <div
      className="switchBagLink"
    >
      <a
        href="/bags"
        id="switchBagLink"
        onClick={[Function]}
      >
        Switch Bags
      </a>
    </div>
  </div>
  <img
    alt="bag"
    className="logo"
    src="bags_bag_handbag_accessory_accessories-19-512.png"
  />
  <h4>
     Estimated Cost for remaining items $
    0
  </h4>
  <button
    onClick={[Function]}
  >
     
    Owned Items
     
  </button>
  <div
    className="itemsList"
  />
  <form
    className="NewItemForm"
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <div
      className="item_name"
    >
      <label
        className="NewItemForm__item_name"
        htmlFor="NewItemForm__item_name"
      >
        Item name
      </label>
      <input
        className="Input "
        id="NewItemForm__item_name"
        name="item_name"
        placeholder="Flashlight"
        required={true}
      />
    </div>
    <div
      className="item_url"
    >
      <label
        className="NewItemForm__item_url"
        htmlFor="NewItemForm__item_url"
      >
        Item Url
      </label>
      <input
        className="Input "
        id="NewItemForm__item_url"
        name="url"
        placeholder="www.amazon.com"
        required={true}
      />
    </div>
    <div
      className="item_img"
    >
      <label
        className="NewItemForm__item_img"
        htmlFor="NewItemForm__item_img"
      >
        Item Image Url
      </label>
      <input
        className="Input "
        id="NewItemForm__item_img"
        name="img"
        placeholder="www.abc@123.com/123"
        required={true}
      />
    </div>
    <div
      className="item_description"
    >
      <label
        className="NewItemForm__item_description"
        htmlFor="NewItemForm__item_description"
      >
        Item Description
      </label>
      <input
        className="Input "
        id="NewItemForm__item_description"
        name="description"
        placeholder="A must have"
        required={true}
      />
    </div>
    <div
      className="item_cost"
    >
      <label
        className="NewItemForm__item_cost"
        htmlFor="NewItemForm__item_cost"
      >
        Item Estimated Cost
      </label>
      <input
        className="Input "
        id="NewItemForm__item_cost"
        name="est_cost"
        placeholder="15.99"
        required={true}
      />
    </div>
    <button
      className="Button submit"
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
